util/src/io/worker.rs

```diff
@@ -123,7 +123,7 @@ impl Worker {
 impl Drop for Worker {
 	fn drop(&mut self) {
 		trace!(target: "shutdown", "[IoWorker] Closing...");
-		let _ = self.wait_mutex.lock();
+		let _ = self.wait_mutex.lock().unwrap();
 		self.deleting.store(true, AtomicOrdering::Release);
 		self.wait.notify_all();
```

Comment:
It has nothing to do with `deleting` as it does not change when the deadlock happens